circuits/scripts/mint.js

```diff
@@ -0,0 +1,101 @@
+// const chalk = require('chalk');
+import { expect }  from 'chai';
+import { default as vmtree } from 'vmtree-sdk';
+
+import { MerkleTree as FixedMerkleTree } from "fixed-merkle-tree";
+
+class MerkleTree extends FixedMerkleTree {
+    constructor({ hasher, levels = 20, leaves = [], zero = 0 }) {
+        super(levels, leaves, {
+            hashFunction: (left, right) => hasher([left, right]),
+            zeroElement: zero,
+        });
+    };
+};
+
+function verifyMerkleProof({pathElements, pathIndices, leaf, root}) {
+    pathElements.forEach((element, i) => {
+        leaf = !pathIndices[i] ?
+            vmtree.poseidon([leaf, element]) : vmtree.poseidon([element, leaf]);
+    });
+    return leaf == root;
+}
+
+const sk1 = '0xb4b0bf302506d14eba9970593921a0bd219a10ebf66a0367851a278f9a8c3d08';
+const pk1 = '0x8ca4cc18dc867aE7D87473f8460120168a895E7A';
+const pk1Uint = 802933809494131860455082925493303288586736066170n;
+
+const sk2 = '0xb81676dc516f1e4dcec657669e30d31e4454e67aa98574eca670b4509878290c';
+const pk2 = '0x2C66bB06B88Bf3aB61aF23E70B0c8bE27b1e5930';
+const pk2Uint = 253486562210967126009990789802080859110172940592n;
+
+const stateTree = new MerkleTree({ hasher: vmtree.poseidon, levels: 3, zero: 0 });
+const mintTree = new MerkleTree({ hasher: vmtree.poseidon, levels: 3, zero: 0 });
+
+// allow a few deposits
+const leaf1 = vmtree.poseidon([pk1Uint, 0, 10]);
+const leaf2 = vmtree.poseidon([pk2Uint, 11, 15]);
```

Comment:
You know, I'd actually be quite happy using this as part of the server. We can start in JS for speed of development and move to Rust if we need it for speed and safety.

In that case we could nicely refactor this into a function that loops over deposits and outputs necessary proofs. So we should move this to its own top level `server` directory.